src/censored.jl

```diff
@@ -68,6 +68,80 @@ quantile(d::Censored, p::Real) = clamp(quantile(d.uncensored, p), d.lower, d.upp
 
 median(d::Censored) = clamp(median(d.uncensored), d.lower, d.upper)
 
+function mean(d::Censored)
+    d0 = d.uncensored
+    lower = d.lower
+    upper = d.upper
+    dtrunc = truncated(d.uncensored, lower, upper)
+    prob_lower = if value_support(typeof(d)) === Discrete && isfinite(lower)
+        cdf(d0, lower) - pdf(d0, lower)
+    else
+        cdf(d0, lower)
+    end
+    prob_upper = ccdf(d0, upper)
+    prob_interval = 1 - (prob_lower + prob_upper)
+
+    μ = prob_interval * mean(dtrunc)
+    if isfinite(lower)
+        μ += prob_lower * lower
+    end
+    if isfinite(upper)
+        μ += prob_upper * upper
+    end
+    return μ
+end
+
+function var(d::Censored)
+    d0 = d.uncensored
+    lower = d.lower
+    upper = d.upper
+    dtrunc = truncated(d.uncensored, lower, upper)
+    prob_lower = if value_support(typeof(d)) === Discrete && isfinite(lower)
```

Comment:
Now this is funny because I was thinking about it, so we use `-Inf::Float64` bounds with discrete ones?

Reply:
Sure, otherwise you introduce an artifical lower or upper bound.

Reply:
The bounds can be of any type for which the cdf can be evaluated. Eg., for Pousson they don't have to be Int and not even integers.

Reply:
We could also use `typemin`, see above

Reply:
This still introduces an undesired artificial bound. Eg. for Poisson distributions only `cdf(d, Inf)` is guarenteed to be 1 but not typemax(Int).

Reply:
I didn't think of that. But it is a tradeoff, right? Because not all `Int` bounds are `Float64` representable

Reply:
Yes, not all can be represented by `Float64`. But this is something we already live with in Distributions but also other packages such as SpecialFunctions in many places: in many functions we convert integers to floating point numbers explicitly with e.g. `float`. So I think it's OK.

Reply:
I just think having separate types for left and right boundary is an easy way out to have `-Inf` to `10000000000000000` as boundary without any troubles.